src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/ApplicationService.kt

```diff
@@ -162,7 +162,14 @@ class ApplicationService(
     }
 
     application.apply {
-      this.data = data
+      this.data = data?.replace("<", "")
+        ?.replace("＜", "")
```

Comment:
ah sorry might not have been clear in my previous comment, I think we would still benefit from a utility function, but it can just take the data string as an argument. something like:

```
  private fun removeXssCharacters(data: String?): String? {
    if (data != null) {
      val xssCharacters = setOf('<', '＜', '〈', '〈', '>', '＞', '〉', '〉')
      var sanitisedData = ""
      xssCharacters.forEach { character ->
        sanitisedData = data.replace(character.toString(), "")
      }
      return sanitisedData
    }
    return null
  }
 ```

which could be called directly from the service

```
    application.apply {
      this.data = removeXssCharacters(application.data)
    }
```

Reply:
Right, so have made these changes Harriet, apart from:

```kotlin
var sanitisedData = data
```
```kotlin
   sanitisedData = sanitisedData?.replace(character.toString(), "")
```
And,
```kotlin
application.apply {
   this.data = removeXssCharacters(data)
}
```
Also changed the parametrised test to test a list of strings (previously chars) and included a test for `<＜〈〈>＞〉〉`:
```kotlin
@ValueSource(strings = ["<", "＜", "〈", "〈", ">", "＞", "〉", "〉", "<＜〈〈>＞〉〉"])
```